src/DocumentDB.ChangeFeedProcessor/Utils/TaskExtensions.cs

```diff
@@ -28,5 +28,23 @@ public static async Task IgnoreException(this Task task)
                 // ignore
             }
         }
+
+        public static async Task<TResult> AbortOnTimeout<TResult>(this Task<TResult> responseTask, TimeSpan timeout, CancellationToken token)
+        {
+            using (CancellationTokenSource cancelSource = CancellationTokenSource.CreateLinkedTokenSource(token))
+            {
+                Task timeoutTask = Task.Delay(timeout, cancelSource.Token);
+
+                if (responseTask == await Task.WhenAny(responseTask, timeoutTask).ConfigureAwait(false))
```

Comment:
If original one is cancelled, that would cause timeoutTask to cancel as well, then WhenAny can trigger the timeout task which would be wrong, right? I think we should just use new cancellation token for timeoutTask.